kadi/tests/test_occweb.py

```diff
@@ -97,3 +96,44 @@ def test_get_fdb_major_events():
 def test_get_fot_major_events():
     page = occweb.get_url('fot_major_events')
     assert 'ACA Dark Current Calibration' in page
+
+
+@pytest.mark.skipif('not HAS_OCCWEB')
+@pytest.mark.parametrize('str_or_Path', [str, Path])
+@pytest.mark.parametrize('cache', [False, 'update', True])
+def test_get_occweb_dir(str_or_Path, cache):
+    """Test get_occweb_dir and get_occweb_page (which is called in the process)"""
+    path = str_or_Path('FOT/mission_planning/PRODUCTS/APPR_LOADS/2000/MAR/')
+    url = f'https://occweb.cfa.harvard.edu/occweb/{path}'
+    files_path = occweb.get_occweb_dir(path, cache=cache)
+    files_url = occweb.get_occweb_dir(url, cache=cache)
+    exp = [
+        '      Name        Last modified   Size',
+        '---------------- ---------------- ----',
+        'Parent Directory               --    -',
+        '       MAR0500D/ 2002-04-30 13:38    -',
+        '       MAR1200D/ 2002-04-30 13:38    -',
+        '       MAR1900E/ 2004-03-18 13:44    -',
+        '       MAR2600C/ 2002-04-30 13:38    -']
+    assert files_path.pformat_all() == exp
+    assert files_url.pformat_all() == exp
+
+
+@pytest.mark.skipif('not HAS_OCCWEB')
+@pytest.mark.parametrize('cache', [False, True])
+def test_get_occweb_dir_fail(cache):
+    """Test get_occweb_dir and get_occweb_page (which is called in the process)"""
+    path = Path('FOT/mission_planning/PRODUCTS/APPR_LOADS/FAIL-NOT-THERE')
+    with pytest.raises(requests.exceptions.HTTPError):
+        occweb.get_occweb_dir(path, cache=cache)
+
+
+@pytest.mark.skipif('not HAS_OCCWEB')
```

Comment:
Should these just have `reason` and match the "modern preferred style"?

Reply:
Indeed, will fix!